test/unit/FieldsetDate.test.js

```diff
@@ -68,8 +71,7 @@ describe('FieldsetDate', () => {
     // FieldsetDate
     const onDateFilterChange = (filterId, value) => {
       noSelectedFilters[filterId] = value;
-
-      return noSelectedFilters;
+      component.setProps({ noSelectedFilters });
```

Comment:
I'm surprised this works. `noSelectedFilters` isn't a prop of `FieldsetDate`. It should be `component.setProps({ selectedFilters: noSelectedFilters })`. Even then, that's not the best test because it's suppose to simulate a change that will internally update the component, but then `setProps` is forcing the change anyway.

Reply:
Good point, it is strange that this works. Normally I would agree that forcing props to change seems like a bad idea, except that in this case the function really updates the parent component, which should then update props. This isn't happening (because there is no parent component) and we need a way to simulate the parent updating the props.

Reply:
Update: can confirm that calling `setProps` will cause the test to work even with no arguments, which supports the theory above. Not sure what the clearest thing to do is between `component.setProps({})`, `component.setProps({ noSelectedFilters })`, and `component.setProps({ selectedFilters: noSelectedFilters })`. I'll go with the last based on your suggestion.